### ISSUE TITLE
tty_fake not working after disconnecting and reconnecting to fake serial port

### ISSUE THREAD

**Issue by AccessArctic** [danielinux/ttybus#9]
After setting correctly a bus with a real serial port and 2 fake port (can see the data in two independent terminal via miniterm/screen/a specific software), if the connection ttyS0fake0 (same with fake1) is closed, it cannot be reopen. It is necessary to kill the tty_fake process and restart it:

    tty_bus -s /tmp/ttyS0mux

on a different terminal:

    tty_attach -s /tmp/ttyS0mux /dev/ttyUSB0

on a different terminal:

    tty_fake -s /tmp/ttyS0mux /tmp/ttyS0fake0 $
    tty_fake -s /tmp/ttyS0mux /tmp/ttyS0fake1

on a different terminal:

    miniterm /tmp/ttyS0fake1 19200

then if closing the miniterm and trying to reopen it with the same command, it won't work

**Comment by nicolaje** [comment]
Hello,

I have the same issue here, tty_fake will exit/close after a process stops reading from it. I don’t know yet if this is intended or if this is an issue that requires fixing.

**Comment by nicolaje** [comment]
Reproductible example:

In terminal 1:
```bash
./tty_bus -s /tmp/ttyS0mux
Creating bus: /tmp/ttyS0mux
```

In terminal 2:
```bash
./tty_fake -s /tmp/ttyS0mux /tmp/ttyS0fake0
Connecting to bus: /tmp/ttyS0mux
Device: /dev/pts/8 is now /tmp/ttyS0fake0
```

In terminal 3:
```bash
cat /tmp/ttyS0fake0 
```

When closing cat in terminal 3, tty_fake in terminal 2 will exit.

@danielinux  is this intended or should I fix this?

**Comment by AccessArctic** [comment]
@nicolaje 
If you can fix it, I would love to have your updated version as for me this is definitely an unwanted behaviour

**Comment by danielinux** [comment]
Is the PULLHUP fix in the pending PR #6  solving this for you? I never had the time to review those. I'm open to merge that PR (and/or more fixups) if it makes sense.

**Comment by AccessArctic** [comment]
Looks promising indeed. I will have a check, but don't have my RPi at hand right now.
Thanks

**Comment by nicolaje** [comment]
Hello, indeed the changes made in this pull request prevent ffy_fake from exiting when a process that was reading from it closes.

So it solves this issue for my use case, although I didn't review the whole PR extensively.

**Comment by nicolaje** [comment]
@danielinux  can I help you further to merge this PR into master ?

**Comment by danielinux** [comment]
Merged #6 . Thanks everyone, apologies for not being very reactive on this project.

**Comment by nicolaje** [comment]
Thank you for your time.
